controllers/managedocs_controller_test.go

```diff
@@ -322,5 +325,213 @@ var _ = Describe("ManagedOCS controller", func() {
 				}, duration, interval).Should(BeTrue())
 			})
 		})
+
+		When("there is an add-on delete comfigmap", func() {
+			ctx := context.Background()
+
+			managedOCS := &v1.ManagedOCS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ManagedOCSName,
+					Namespace: TestNamespace,
+				},
+			}
+			sc := &ocsv1.StorageCluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      storageClusterName,
+					Namespace: TestNamespace,
+				},
+			}
+
+			storageClassRbdName := "ocs-storagecluster-ceph-rbd"
+			storageClassCephFsName := "ocs-storagecluster-cephfs"
+
+			pvc1 := &corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pvc1",
+					Namespace: TestNamespace,
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Gi"),
+						},
+					},
+					StorageClassName: &storageClassRbdName,
+				},
+			}
+			pvc2 := &corev1.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-pvc2",
+					Namespace: TestNamespace,
+				},
+				Spec: corev1.PersistentVolumeClaimSpec{
+					AccessModes: []corev1.PersistentVolumeAccessMode{corev1.ReadWriteOnce},
+					Resources: corev1.ResourceRequirements{
+						Requests: corev1.ResourceList{
+							corev1.ResourceStorage: resource.MustParse("1Gi"),
+						},
+					},
+					StorageClassName: &storageClassCephFsName,
+				},
+			}
+			sub := &operators.Subscription{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      TestAddonSubscriptionName,
+					Namespace: TestNamespace,
+				},
+				Spec: &operators.SubscriptionSpec{},
+			}
+			csv := &operators.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      TestdeployerCsvName,
+					Namespace: TestNamespace,
+				},
+				Spec: operators.ClusterServiceVersionSpec{
+					InstallStrategy: operators.NamedInstallStrategy{
+						StrategyName: "test-strategy",
+						StrategySpec: operators.StrategyDetailsDeployment{
+							DeploymentSpecs: []operators.StrategyDeploymentSpec{},
+						},
+					},
+				},
+			}
```

Comment:
[suggestion] I like that there are nested whens for the different cases. Since these objects are being used only within the nested when's scope, I think it would be better for readibility to move initialize these objects there.

Reply:
Moved sub and csv objects to respective when block. As pvc objects are used in more than when conditions, i prefer to keep them here.